### DISCUSSION TITLE
htpassword access

### DISCUSSION THREAD

**Discussion by vulkanosaure** [TimWolla/docker-adminer#142] [Q&A]
I'm trying to setup a htpasswd access (the same way apache allows), to have an additional security layer
I looked into the container searching how to add this myself but i have no clue (i don't see what webserver is being used here)

any hints ? 
cheers

**Comment by TimWolla** [comment]
The standalone version uses PHP's builtin web server, which doesn't even do HTTPS. It's intended for use within your local development environment for that reason alone.

Configure authentication within your web server or reverse proxy if you want to make the container available to the Internet – and make sure to also set-up TLS encryption.

**Reply by vulkanosaure** [reply]
So I have a docker container running this adminer image on my prod server. Do you mean I should setup the authentication on the host level web server ?

**Reply by TimWolla** [reply]
> Do you mean I should setup the authentication on the host level web server ?

In whatever software you put in front of the container, yes. If you are directly exposing the container to the Internet, stop, because that is unsafe. For a production setup you might want to prefer the FastCGI variant. There's nginx config examples somewhere in the existing issues / discussions.
